File: PluginMessagingUtils/src/main/java/org/openredstone/executors/KillExecutor.java
package org.openredstone.executors;

import org.bukkit.entity.Player;
import org.bukkit.plugin.Plugin;
import org.openredstone.messages.ActionMessage;

public class KillExecutor extends Executor {

    public KillExecutor(Plugin plugin) {
        super(plugin);
    }

    @Override
    public void execute(ActionMessage actionMessage) throws Exception {
        Player player = plugin.getServer().getPlayer(actionMessage.getUuid());

        if (player == null) {
            throw new Exception("Player not found.");
        }

        player.setHealth(0);
    }
}

File: plugin-messaging/src/main/kotlin/org/openredstone/executors/KillExecutor.kt
package org.openredstone.executors

import org.bukkit.entity.Player
import org.bukkit.plugin.Plugin
import org.openredstone.messages.ActionMessage

class KillExecutor(plugin: Plugin) : Executor(plugin) {
    override fun execute(actionMessage: ActionMessage) {
        val player: Player =
            plugin.server.getPlayer(actionMessage.uuid)
                ?: throw Exception("Player not found.")
        player.health = 0.0
    }
}
